aip/auth/4110.md

```diff
@@ -136,25 +136,28 @@ digraph d_front_back {
     1. Otherwise, get an access token _[END]_
   1. If false, go to step (2.3)
 1. **Determine auth flows**
-  1. If the credential is gcloud credential go to step (5.3)
-  1. If target audience or scope is provided by the developer go to step (5.1)
-  1. Otherwise, go to step (5.2)
+  1. If the credential is a [gcloud credential JSON](5) go to step (5.3)
+  1. If the credential is a [service account key JSON](6):
+     1. If neither `scope` or `audience` is available, return an error _[END]_
+     1. If `audiences` are **explicitly** provided by developer, which indicates that the developer asks to use self-signed JWT, go to step (5.1)
+     1. If `audiences` are **not explicitly** provided:
+        1. If scope is available and [scope is **not supported** for self-signed JWT](8), go to step (5.2)
+        1. Otherwise, go to step (5.1)
 1. **Execute auth flows**
-  1. Use 2LO flow to exchange for an auth token
-    1. If target audience is provided by the developer, get an [identity token][7] _[END]_
-    1. Otherwise, get an access token _[END]_
-      1. If client certificate is presented, the exchanged token will be a certificate bind token. _[END]_
-  1. Use self-signed JWT flow to create an access token locally.
-    1. If certificate is presented, embed the certificate into the JWT.
-    1. Use the regular [self-signed JWT flow][4] for an access token. _[END]_
-  1. Use user identity flow to exchange for an access token. _[END]_
+  1. Use the [self-signed JWT flow][4] for an access token.
+     1. If `audiences` are **explicitly** provided by developer, create the JWT with the `audience`
+     1. If [scope is supported for self-signed JWT](8) and `scope` is provided, create the JWT with the `scope`.
+     1. Otherwise, create the JWT with `audience` _[END]_
```

Comment:
Can you clarify whether this case 3 refers to something like a "default audience"? Otherwise, it would appear to be superseded by case 1.